symbolic-demangle/src/lib.rs

```diff
@@ -459,4 +470,24 @@ mod test {
             md5_mangled
         );
     }
+
+    #[test]
+    fn test_strip_hash_suffix() {
+        assert_eq!(
+            strip_hash_suffix("hello$0123456789abcdef0123456789abcdef"),
+            "hello"
+        );
+        assert_eq!(
+            strip_hash_suffix("hello_0123456789abcdef0123456789abcdef"),
+            "hello_0123456789abcdef0123456789abcdef",
+        );
+        assert_eq!(
+            strip_hash_suffix("hello\u{1000}0123456789abcdef0123456789abcdef"),
+            "hello\u{1000}0123456789abcdef0123456789abcdef"
+        );
+        assert_eq!(
+            strip_hash_suffix("hello$0123456789abcdef0123456789abcdxx"),
+            "hello$0123456789abcdef0123456789abcdxx"
+        );
```

Comment:
nit: could i bother you to add a unit test with a 2-byte character that comes after the `$` to check for the scenario where a multibyte char bumps the iterator past the expected index?